### PR TITLE
tests: subsys: Adding testcases to check tracing APIs

### PR DESCRIPTION
**Pull request by Zhaoningx**
Add some testcases to verify tracing's APIs,
check output data by using uart backend of self-design.

Signed-off-by: Zhaoningx <ningx.zhao@intel.com>

### CODE REVIEW

File: tests/subsys/tracing/tracing_api/src/main.c [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+#include <logging/log.h>
+#include <tracing_buffer.h>
+#include <tracing_core.h>
+#include <tracing/tracing_format.h>
+#if defined(CONFIG_TRACING_BACKEND_UART)
+#include "../../../../subsys/tracing/include/tracing_backend.h"
+#endif
+
+/* size of stack area used by each thread */
+#define STACK_SIZE (1024 + CONFIG_TEST_EXTRA_STACKSIZE)
```
Comment (LeiW000) [inline tests/subsys/tracing/tracing_api/src/main.c:18]: It is not used in the code. I think we can remove it.

File: tests/subsys/tracing/tracing_api/src/main.c [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+	sys_trace_k_thread_abort_exit(&thread);
+	sys_trace_k_thread_yield();
+	sys_trace_k_thread_wakeup(&thread);
+	sys_trace_k_thread_pend(&thread);
+	sys_trace_k_thread_info(&thread);
+	sys_trace_k_thread_name_set(&thread, ret);
+	sys_trace_k_thread_sched_lock();
+	sys_port_trace_k_thread_sched_unlock();
+	sys_trace_k_thread_join_blocking(&thread, timeout);
+	sys_trace_k_thread_join_exit(&thread, timeout, ret);
```
Comment (LeiW000) [inline tests/subsys/tracing/tracing_api/src/main.c:116]: It seems to me that "thread_api_found" will be set if any of the sys_trace_k_thread_* is called successfully. I don't think it is correct.

**Review by LeiW000 — CHANGES_REQUESTED** [review]

**Review by enjiamai — CHANGES_REQUESTED** [review]
Hi @Zhaoningx  Could you please add the detail and brief of these testcases? I think it is necessary.

**Comment by Zhaoningx** [comment]
> Hi @Zhaoningx Could you please add the detail and brief of these testcases? I think it is necessary.

Hi @enjiamai , I have added the description, could you like to review this, thanks.

File: tests/subsys/tracing/tracing_api/src/main.c (hunk 2 above)
Comment (Zhaoningx) [reply tests/subsys/tracing/tracing_api/src/main.c:149]: Hi @LeiW000 , fixed, could you like to review this, thanks.

File: tests/subsys/tracing/tracing_api/testcase.yaml [hunk 3]
```diff
@@ -0,0 +1,10 @@
+tests:
+  tracing.transport.uart.async.test:
+    platform_allow: qemu_x86
+    extra_args: CONF_FILE="prj_uart.conf"
+  tracing.transport.uart.sync.test:
+    platform_allow: qemu_x86
+    extra_args: CONF_FILE="prj_uart.conf"
+    extra_configs:
+      - CONFIG_TRACING_SYNC=y
+      - CONFIG_TRACING_ASYNC=n
```
Comment (chen-png) [inline tests/subsys/tracing/tracing_api/testcase.yaml:10]: if you set CONFIG_TRACING_SYNC as y, don't need to set CONFIG_TRACING_ASYNC

File: tests/subsys/tracing/tracing_api/testcase.yaml [hunk 4]
```diff
@@ -0,0 +1,10 @@
+tests:
+  tracing.transport.uart.async.test:
+    platform_allow: qemu_x86
+    extra_args: CONF_FILE="prj_uart.conf"
```
Comment (chen-png) [inline tests/subsys/tracing/tracing_api/testcase.yaml:4]: I think this is a common file, you can rename prj_uart.conf as prj.conf, then don't need to use extra_args

Comment (Zhaoningx) [reply tests/subsys/tracing/tracing_api/testcase.yaml]: Hi @chen-png , I have updated the code, thanks.

**Review by LeiW000 — APPROVED** [review]

**Review by enjiamai — APPROVED** [review]

**Pull request merged**